cpp/src/arrow/acero/aggregate_internal.h

```diff
@@ -143,11 +143,10 @@ Status HandleSegments(RowSegmenter* segmenter, const ExecBatch& batch,
 }
 
 /// @brief Extract values of segment keys from a segment batch
-/// @param[out] values_ptr Vector to store the extracted segment key values
+/// @param[out] values Vector to store the extracted segment key values
 /// @param[in] input_batch Segment batch. Must have the a constant value for segment key
 /// @param[in] field_ids Segment key field ids
-Status ExtractSegmenterValues(std::vector<Datum>* values_ptr,
-                              const ExecBatch& input_batch,
+Status ExtractSegmenterValues(std::vector<Datum>& values, const ExecBatch& input_batch,
```

Comment:
Hmm, could we keep the convention of avoiding mutable refs in APIs (even internal)? They help seeing _at the call site_ that some variable is going to be mutated.

Reply:
Oops, didn't realize this is by coding convention. Updated. Thanks!